os/fs/smartfs/smartfs_smart.c

```diff
@@ -186,37 +186,48 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 	/* Get the mountpoint inode reference from the file structure and the
 	 * mountpoint private data from the inode structure
 	 */
-
 	inode = filep->f_inode;
 	fs = inode->i_private;
 
 	DEBUGASSERT(fs != NULL);
 
-	/* Take the semaphore */
+	/* If the file has not been opened for writing but an Append or Truncate has
+	 * been requested, return an error
+	 */
+	/* TODO: Do we allow a file to be Created in READ mode only i.e. (O_READ | O_CREAT)?
+	 * 	The file has no data to be read yet, so we will only be able to create an empty file entry.
+	 */
+	if ((oflags & O_WROK) == 0 && (oflags & (O_TRUNC | O_APPEND))) {
+		fdbg("Invalid file-opening mode for %s\n", relpath);
```

Comment:
When we say invalid mode, how about printing the mode value?

Reply:
Printing the mode value as a number I feel would be insufficient for the user/application programmer. Additionally, this is kind of an extra check for water tightness.